cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeUtils.java

```diff
@@ -84,6 +84,10 @@
   public static final String PYSPARK_PORT_FILE_NAME = "cdap.py4j.gateway.port.txt";
   public static final String PYSPARK_SECRET_FILE_NAME = "cdap.py4j.gateway.secret.txt";
 
+  public static final String STREAMING_CHECKPOINT_REWRITE_ENABLED = "streaming.checkpoint.rewrite.enabled";
+  public static final String SPARK_STREAMING_CHECKPOINT_REWRITE_ENABLED =
```

Comment:
Why need two separate constants? Can we just have one? Also, since the constant is defined inside spark, it doesn't need to be so long. E.g. It can be just `CHECKPOINT_REWRITE_ENABLED`.

Reply:
I made them separate as I use the `STREAMING_CHECKPOINT_REWRITE_ENABLED` variable for the Java options in https://github.com/cdapio/cdap/pull/12822/files#diff-25b96ab36cd957bb35aace06ff7a5befcc138b4900b04c2e1435951a252a51b6R532

The `SPARK_STREAMING_CHECKPOINT_REWRITE_ENABLED` just adds the `spark.` prefix and is used in the `cdap-site.xml` configuration file.

Reply:
Seems like you can just use the same one?

Reply:
I tried that and when I did, Spark complained that Spark configuration properties cannot be set using Java opts.

I don't recall the exact text of the error message, I can reproduce it and paste it here if you'd like.

Reply:
I see. it's ok to keep this. Maybe good to add a code comment as to why it is needed.